qemu/tests/memory_leak_after_nichotplug.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import re
+import time
+
+from avocado.utils import process
+from virttest import error_context
+from virttest import utils_net
+from virttest import utils_misc
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory leak check after nic hotplug/hotunplug
+    1) Boot a guest
+    2) Check free memory
+    3) Hotplug nic 100 times(windows) or add 300 vlan(linux)
+    4) Hotunplug nic 100 times(windows) or del 300 vlan(linux)
+    5) Check free memory again
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def get_ovs_ports(ovs):
+        cmd = "ovs-vsctl list-ports %s" % ovs
+        return process.system_output(cmd, shell=True).decode()
+
+    def is_ovs_backend(netdst):
+        return netdst in process.system_output("ovs-vsctl list-br",
+                                               ignore_status=True,
+                                               shell=True).decode()
+
+    def cleanup_ovs_ports(netdst, ports):
+        if is_ovs_backend(netdst) is True:
+            ports = set(get_ovs_ports(netdst).splitlines()) - \
+                set(ports.splitlines())
+            for p in ports:
+                process.system("ovs-vsctl del-port %s %s" % (netdst, p))
+
+    netdst = params.get("netdst", "switch")
+    host_hw_interface = get_ovs_ports(netdst)
```

Comment:
@xiywang Seems this is not a ovs backend only case. For bridge backend, test will fail:

ERROR: Command 'ovs-vsctl list-ports switch' failed.\nstdout: b''\nstderr: b'ovs-vsctl: unix:/var/run/openvswitch/db.sock: database connection failed (No such file or directory)\n'\nadditional_info: None